proto/redpanda/api/dataplane/v1alpha1/transform.proto

```diff
@@ -0,0 +1,206 @@
+syntax = "proto3";
+
+package redpanda.api.dataplane.v1alpha1;
+
+import "buf/validate/validate.proto";
+import "google/api/annotations.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+message TransformMetadata {
+  string name = 1;
+  string input_topic_name = 2;
+  repeated string output_topic_names = 3;
+  repeated PartitionTransformStatus status = 4;
```

Comment:
We should maybe put some validation annotations in this message? At least the topic names can be validated like:
```proto
string input_topic_name = 2 [
      (buf.validate.field).required = true,
      (buf.validate.field).string.min_len = 1,
      (buf.validate.field).string.max_len = 249,
      (buf.validate.field).string.pattern = "^[a-zA-Z0-9._\\-]*$"
    ];
```

Reply:
I don't think it makes sense to validate TransformMetadata as it is purely an output element. 

I do have validation on fields on incoming elements though

Reply:
Well, except for these two on DeployRequest that were pending me copying the validation rules from Topic lol 


// 'input_topic_name' specifies the name of the input topic for the transform.
  string input_topic_name = 3 [
    (buf.validate.field).string.min_len = 1,
    (buf.validate.field).string.max_len = 255,
    (buf.validate.field).string.pattern = "^[a-z0-9.]+$"
  ];

  // 'output_topic_names' is a list of output topics for the transform. This field can contain
  // multiple strings, each representing the name of an output topic.
  repeated string output_topic_names = 4 [
    (buf.validate.field).string.min_len = 1,
    (buf.validate.field).string.max_len = 255,
    (buf.validate.field).string.pattern = "^[a-z0-9.]+$"
  ];